src/dagmc/dagmcmetadata.cpp

```diff
@@ -0,0 +1,591 @@
+        implicit_complement_material = material_props[position].substr(0, material_props[position].size() - 5);
+        implicit_complement_density = density_props[0];
+        material_props.erase(material_props.begin() + position);
+      } else {
+        // failure a volume can only have a single material associated with it
+        std::cout << "more than one material for volume with id " << cellid << std::endl;
+        std::cout << "that does not the the _comp tag associated with it" << std::endl;
+        std::cout << cellid << " has the following material assignments" << std::endl;
+        for (int j = 0 ; j < material_props.size() ; j++) {
+          std::cout << material_props[j] << std::endl;
+        }
+        std::cout << "Please check your material assignments " << cellid << std::endl;
+        exit(EXIT_FAILURE);
+      }
+    } else {
+      // because of how the data are inserted into the map, ther are always
+      // at least one entry, "" is nothing is found
+      // if there is no material property - not failure for impl_comp
+      if (material_props[0] == "" && !(DAG->is_implicit_complement(eh))) {
+        std::cout << "No material property found for volume with ID " << cellid << std::endl;
+        std::cout << "Every volume must have only one mat: property" << std::endl;
+        exit(EXIT_FAILURE);
+      }
+    }
+
+    // this is never ok for a volume to have more than one proprety for density
+    if (density_props.size() > 1) {
+      std::cout << "More than one density specified for " << cellid << std::endl;
+      std::cout << cellid << " has the following density assignments" << std::endl;
+      for (int j = 0 ; j < density_props.size() ; j++) {
+        std::cout << density_props[j] << std::endl;
+      }
+      std::cout << "Please check your density assignments " << cellid << std::endl;
+      exit(EXIT_FAILURE);
+    }
+
+    std::string grp_name = "";
+
+    // determine if we have a density property
+    if (!density_props[0].empty()) {
+      grp_name = "mat:" + material_props[0] + "/rho:" + density_props[0];
+      volume_density_data_eh[eh] = density_props[0];
+    } else {
+      grp_name = "mat:" + material_props[0];
+      volume_density_data_eh[eh] = "";
+    }
+
+    // check to see if the simplied naming scheme is used, by try to convert the
+    // material property to an int
+    if (try_to_make_int(material_props[0]) && density_props[0].empty() && !(DAG->is_implicit_complement(eh))) {
+      std::cout << "Using the simplified nameing scheme without a density" << std::endl;
+      std::cout << "property is forbidden, please rename the group mat:" << material_props[0] << std::endl;
+      exit(EXIT_FAILURE);
+    }
+
+    // set the material value
+    volume_material_property_data_eh[eh] = grp_name;
+
+    // not graveyard or vacuum or implicit compliment
+    if (grp_name.find("Graveyard") == std::string::npos && grp_name.find("Vacuum") == std::string::npos
```

Comment:
Reducing future errors: May make sense to define "Graveyard" and "Vacuum" somewhere as constants, rather than using the strings directly here and on line 198 below.  Centralizing the string in a constant rather than using "magic strings" so-to-speak stops subtle bugs from potential misspellings later.